### ISSUE TITLE
[Bug] macro prebuild is not triggered in incremental installation + wrong macro binary path in Xcode 16

### ISSUE THREAD

**Issue by wangzhizhou** [trinhngocthuyen/cocoapods-spm#135]
### What happened?

# Problem 1

Steps:
1. change the Podfile of examples as follow:
```ruby 
  config_cocoapods_spm(
    dont_prebuild_macros: false,                                                                                                                            
    default_macro_config: "debug"
  )
```
2.  run the command: `make ex.install`
3. check the directory: `.spm.pods/macros/.downloaded/MacroCodableKit/`, 
4. there is no `.build` directory, 

Result: 
1. The prebuild action is not triggered by default.
2. The plugin dose not call the `prebuild_macro_impl` method by default.

# Problem 2

Steps:
1. I execute the prebuild action mannually by run command: 
    ```bash
    bundle exec pod spm macro prebuild --macros=MacroCodableKit --config=debug
    ```
2. check the directory: `.spm.pods/macros/.downloaded/MacroCodableKit/.build/debug`
4. the binary product is  `<product_name>-tool`, and can not be copyed as the prebuild binary by plugin:
![image](https://github.com/user-attachments/assets/4b243b4c-c2d7-43e6-bd5d-cd5e35acba19)
![image](https://github.com/user-attachments/assets/17ca15da-725f-4459-8316-0a39ce9bbfbb)
the prebuild product named with the '-tool' suffix, which maybe caused by swift 6 different compiled rule?
![image](https://github.com/user-attachments/assets/221f199c-da2a-4785-9502-39add2e0e3cb)

### Stack

```
   CocoaPods : 1.16.2
        Ruby : ruby 3.3.6 (2024-11-05 revision 75015d4c1f) [arm64-darwin24]
    RubyGems : 3.5.23
        Host : macOS 15.1.1 (24B91)
       Xcode : 16.0 (16A242d)
         Git : git version 2.46.1
Ruby lib dir : /opt/homebrew/Cellar/ruby/3.3.6/lib
d8710554b4ebfd1a6a8bac6c41cb1fc3f1dc8e8f
               trunk - CDN - https://cdn.cocoapods.org/
```

### Installation Source

```
Executable Path: /opt/homebrew/lib/ruby/gems/3.3.0/bin/pod
```

### Plugins

```
cocoapods-compact-spec   : 0.0.4
cocoapods-deintegrate    : 1.0.5
cocoapods-plugins        : 1.0.0
cocoapods-search         : 1.0.1
cocoapods-spm            : 0.1.8
cocoapods-trunk          : 1.6.0
cocoapods-try            : 1.2.0
cocoapods-xcconfig-hooks : 0.0.1
```

### Podfile

```ruby
# --------------------------------------------------
# Base configs
# --------------------------------------------------
@checksum = "dummy-checksum-to-prevent-merge-conflicts"
ios_deployment_target = "16.0"

platform :ios, ios_deployment_target
use_frameworks! :linkage => (ENV["LINKAGE"] || :static).to_sym
install! "cocoapods", :generate_multiple_pod_projects => true

# --------------------------------------------------
# Plugin configs
# --------------------------------------------------
plugin "cocoapods-spm"
plugin "cocoapods-xcconfig-hooks"
config_compact_spec(
  extra: ->(s) { s.ios.deployment_target = ios_deployment_target }
)

config_cocoapods_spm(
  dont_prebuild_macros: false,
  default_macro_config: "debug"
)

# --------------------------------------------------
# Pods config
# --------------------------------------------------

def shared_pods
  pod "Logger", :path => "LocalPods/Logger"
end

target "EX" do # rubocop:disable Metrics/BlockLength
  shared_pods
  pod "CommonUI", :path => "LocalPods/CommonUI"
  pod "Services", :path => "LocalPods/Services"

  pod "Orcam", :macro => {
    :git => "https://github.com/trinhngocthuyen/orcam.git",
    :branch => "main",
  }
  pod "MacroCodableKit", :macro => {
    :git => "https://github.com/mikhailmaslo/macro-codable-kit",
    :tag => "0.3.0",
  }

  spm_pkg "SnapKit",
          :url => "https://github.com/SnapKit/SnapKit.git",
          :version => "5.7.1",
          :products => ["SnapKit-Dynamic"]
  spm_pkg "SwiftUIX", :git => "https://github.com/SwiftUIX/SwiftUIX.git", :tag => "0.1.9"
  spm_pkg "SwiftyBeaver", :git => "https://github.com/SwiftyBeaver/SwiftyBeaver.git", :tag => "2.0.0"
  spm_pkg "opentelemetry-swift",
          :git => "https://github.com/open-telemetry/opentelemetry-swift.git",
          :branch => "main",
          :products => ["OpenTelemetrySdk"]
  spm_pkg "GoogleMaps",
          :git => "https://github.com/googlemaps/ios-maps-sdk.git",
          :version => "8.4.0",
          :products => ["GoogleMaps", "GoogleMapsBase", "GoogleMapsCore"]
  spm_pkg "mqtt-nio",
          :url => "https://github.com/swift-server-community/mqtt-nio.git",
          :version => "2.11.0",
          :products => ["MQTTNIO"]
  spm_pkg "DebugKit", :path => "LocalPackages/debug-kit"
end

target "EXTests" do
  # Nothing here
end

target "EXMac" do
  platform :osx, "14.0"
  shared_pods
end

post_install do |installer|
  installer.native_targets.each do |target|
    target.build_configurations.each do |config|
      config.build_settings["IPHONEOS_DEPLOYMENT_TARGET"] = ios_deployment_target
    end
  end
end
```

### Anything else?

CocoaPods 1.16.2 + Ruby 3.2.2 + Xcode 16.0 + Swift 6 + macOS 15.1.1

**Comment by trinhngocthuyen** [comment]
### For problem (1)
Found the root cause:
- The macro prebuild logic is triggered together with `installer.install!` https://github.com/trinhngocthuyen/cocoapods-spm/blob/main/lib/cocoapods-spm/macro/pod_installer.rb#L10-L13
- However, CocoaPods has the cache to skip PodSourceInstaller for incremental installation. Let say, if we delete `Pods` dir, or just `Pods/Local Podspecs/MacroCodableKit.podspec.json`, the prebuild will be triggered again. But the subsequent runs will not.

-> I think we can force-run this step for macro pods in incremental installations.

###  For problem (2)
It turns out Xcode 16 changes the macro binary format, having a `-tool` suffix in the name 😅.
![macro_xc1516](https://github.com/user-attachments/assets/c0aa8f2c-1d9b-4f2a-8a54-f67f30ca9ca0)

**Comment by wangzhizhou** [comment]
for Problem 2, maybe we should found another way to locate the correct binary name

**Comment by wangzhizhou** [comment]
For problem 1, I'm not sure about the caching mechanism of the installation at the moment, and it may take a little more time to understand it

**Comment by trinhngocthuyen** [comment]
Problem 1 has been fixed in #137. Could you help check it out. Thanks!

**Comment by wangzhizhou** [comment]
Ok, great job.

**Comment by wangzhizhou** [comment]
I just test the fix #137 in my project, and it works, when the pod install executed, the prebuild binary be produced and integerated into the build settings successfully. Thanks for your great job. 👏
